### PR TITLE
feat: support blobfuse-proxy on OpenShift

### LINKED ISSUE #791: blobfuse-proxy support core os (OpenShift)
**Is your feature request related to a problem?/Why is this needed**

Currently blobfuse-proxy is only available on debian based agent node, it's not supported on core os (OpenShift)

https://github.com/kubernetes-sigs/blob-csi-driver/tree/master/deploy/blobfuse-proxy

**Describe the solution you'd like in detail**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by cvvz**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
support blob-csi-driver in openshift
**Which issue(s) this PR fixes**:

Fixes #791

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:
in coreos, we could just copy the blobfuse2 binary to /usr/local/bin/blobfuse2

**Release note**:
```
support blob-csi-driver in openshift
```

### CODE REVIEW

File: pkg/blobfuse-proxy/blobfuse-proxy.service
Comment (andyzhangx) [inline pkg/blobfuse-proxy/blobfuse-proxy.service:5]: since this path has been changed, what's the impact if user upgrades from old version to this version?

File: pkg/blobfuse-proxy/init.sh
Comment (andyzhangx) [inline pkg/blobfuse-proxy/init.sh:109]: if blobfuse-proxy is still running on Ubuntu, will `rm -rf` operation fail?

**Review by andyzhangx — COMMENTED** [review]
since there are a few different behaviors between Ubuntu and OpenShift, e.g.
1. blobfuse-proxy path
2. blobfuse install
3. other os related env changes

I think it's better making separate init.sh and blobfuse-proxy.service for Ubuntu and CoreOS, that would version upgrade as little risky as possible.

**Comment by cvvz** [comment]
I've split the installation script into two different scripts for different Linux dstros. Please take another look. @andyzhangx  @jsafrane

File: pkg/blobfuse-proxy/install-proxy-rhcos.sh
Comment (andyzhangx) [inline pkg/blobfuse-proxy/install-proxy-rhcos.sh:19]: @jsafrane could you take a look at this script? thanks.

File: pkg/blobfuse-proxy/install-proxy-rhcos.sh
Comment (andyzhangx) [inline pkg/blobfuse-proxy/install-proxy-rhcos.sh:19]: add a more detailed comments here: 
```
in coreos, we could just copy the blobfuse2 binary to /usr/local/bin/blobfuse2

```

File: pkg/blobfuse-proxy/install-proxy-rhcos.sh
Comment (andyzhangx) [inline pkg/blobfuse-proxy/install-proxy-rhcos.sh:20]: if we only support blobfuse2 install, then we need to make some changes here for coreos?

https://github.com/kubernetes-sigs/blob-csi-driver/blob/b99e1cf51f7f3dfcccd9dae77c74de5cdab8f6bc/pkg/blobfuse-proxy/server/server.go#L124-L127

Comment (cvvz) [reply pkg/blobfuse-proxy/install-proxy-rhcos.sh:20]: Nice catch!

File: pkg/blobfuse-proxy/install-proxy-rhcos.sh
Comment (jsafrane) [inline pkg/blobfuse-proxy/install-proxy-rhcos.sh:3]: Nit: 2024

File: charts/latest/blob-csi-driver/templates/csi-blob-node.yaml
Comment (jsafrane) [inline charts/latest/blob-csi-driver/templates/csi-blob-node.yaml:104]: Is this necessary if you have already `/host/usr` ?

**Review by jsafrane — APPROVED** [review]
From what I can tell without testing, this looks good for RHCOS.

**Comment by rh-aelgendy** [comment]
Hi @feiskyer, It seems that the PR is pending your approval currently, please let us know if any further contribution is needed from Red Hat end in order to successfully proceed with the testing.

**Comment by andyzhangx** [comment]
> Hi @feiskyer, It seems that the PR is pending your approval currently, please let us know if any further contribution is needed from Red Hat end in order to successfully proceed with the testing.

@rh-aelgendy this PR has conflict now, weizhi is on vacation, we will try to merge this PR next week when weizhi is back.

File: charts/latest/blob-csi-driver/templates/csi-blob-node.yaml
Comment (cvvz) [reply charts/latest/blob-csi-driver/templates/csi-blob-node.yaml:104]: Yes, for coreOS, /usr/ and /usr/local are different mount point.

Comment (andyzhangx) [reply charts/latest/blob-csi-driver/templates/csi-blob-node.yaml:104]: I think what @jsafrane mean is that since L102 is already mounting with `/usr`, why not resuse this mount point

Comment (cvvz) [reply charts/latest/blob-csi-driver/templates/csi-blob-node.yaml:104]: Becase for coreOS, /usr/local is not a subpath of /usr, it a different mount point, so we have to mount /usr/local and /usr seperately.

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by andyzhangx** [comment]
@cvvz could you cherrypick to release-1.23 manually?

**Pull request merged**
